### PR TITLE
feat(query): add cbs disk without encrypted for tencentcloud

### PR DESCRIPTION
**Pull request by hellertang**
**Proposed Changes**
- CBS Disk Without Encrypted for TencentCloud

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

**Comment by hellertang** [comment]
Already add new cloudProvider : `tencentcloud` in `.github/scripts/queries-validator/metadata-schema.json`. Please check and run again.  Thanks.

File: assets/queries/terraform/tencentcloud/disk_encryption_disabled/metadata.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "queryName": "Disk Encryption Disabled",
+  "severity": "MEDIUM",
+  "category": "Encryption",
+  "descriptionText": "Disks should have encryption enabled",
+  "descriptionUrl": "https://registry.terraform.io/providers/tencentcloudstack/tencentcloud/latest/docs/resources/cbs_storage#encrypt",
+  "platform": "Terraform",
+  "descriptionID": "1ee0f202",
+  "cloudProvider": "tencentcloud",
+  "cwe": ""
+}
```
Comment (gabriel-cx) [inline assets/queries/terraform/tencentcloud/disk_encryption_disabled/metadata.json:12]: Kindly add `"experimental": "true"` in this `metadata.json` file.
[Check this as an example](https://github.com/Checkmarx/kics/blob/master/assets/queries/terraform/nifcloud/computing_instance_has_common_private/metadata.json#L11).

**Review by gabriel-cx — CHANGES_REQUESTED** [review]

**Comment by hellertang** [comment]
Why does the e2e-debian-tests fail? I can’t seem to see the detailed error message.
Please tell me how to fix.
Thanks.

**Comment by gabriel-cx** [comment]
> Why does the e2e-debian-tests fail? I can’t seem to see the detailed error message. Please tell me how to fix. Thanks.

Hi @hellertang ,

In this case we just need to rerun the test. The issue is not related to your PR's changes, do not worry :D

**Comment by hellertang** [comment]
> > Why does the e2e-debian-tests fail? I can’t seem to see the detailed error message. Please tell me how to fix. Thanks.
> 
> Hi @hellertang ,
> 
> In this case we just need to rerun the test. The issue is not related to your PR's changes, do not worry :D

Thanks.  We sincerely hope that KICS can support the inspection of Tencent Cloud's Terraform resources. I aspire for this PR to be integrated, as a testament to the feasibility of this path. In doing so, we can strive to contribute more inspection rules pertinent to the Tencent Cloud Provider.

**Comment by gabriel-cx** [comment]
> > > Why does the e2e-debian-tests fail? I can’t seem to see the detailed error message. Please tell me how to fix. Thanks.
> > 
> > 
> > Hi @hellertang ,
> > In this case we just need to rerun the test. The issue is not related to your PR's changes, do not worry :D
> 
> Thanks. We sincerely hope that KICS can support the inspection of Tencent Cloud's Terraform resources. I aspire for this PR to be integrated, as a testament to the feasibility of this path. In doing so, we can strive to contribute more inspection rules pertinent to the Tencent Cloud Provider.

Nice to know! Thank you again for this amazing contribution!

Kindly be aware that, since this is a new _cloud provider support (tencentcloud)_, in order to run this new queries you need to use the KICS flag `--experimental-queries`.
After merging this PR, we will update our documentation as well, so we can mention Tencent Cloud as a new support, under our [Beta Features section.](https://github.com/Checkmarx/kics?tab=readme-ov-file#beta-features)

**Comment by gabriel-cx** [comment]
@hellertang kindly add a line for tencentcloud [here](https://github.com/Checkmarx/kics/blob/master/test/main_test.go#L43) as well. So the unit tests can run against tencentcloud queries.

**Comment by gabriel-cx** [comment]
thanks @hellertang !

kindly be aware of this failing unit test:
image

**Comment by hellertang** [comment]
> thanks @hellertang !
> 
> kindly be aware of this failing unit test: image

Thanks @gabriel-cx . Already fix it just now.

I have discerned that if 'experimental' is set to true, 'Remediation' cannot be established in 'query.go', as this would result in the unit test not being able to execute successfully. So I remove it.

**Review by gabriel-cx — APPROVED** [review]

**Comment by gabriel-cx** [comment]
Thank you @hellertang !
This week we will release a new KICS version (v1.7.13) and Tencent Cloud support will be part of it!

**Comment by gabriel-cx** [comment]
Hi @hellertang  ,

I'm happy to let you know that your tencentcloud queries are now part of the KICS 'official queries'!
This means, tencentcloud queries will run by default when KICS detects terraform files.
This functionality will start to be available in the next KICS version (v2.1.0).

Thank you again for all your help on this!

**Pull request merged**